src/main/webapp/profileScript.js

```diff
@@ -107,13 +108,37 @@ function addRoute(newRoute) {
   routeRating.classList.add("rating");
 
   routeDetails.innerHTML = newRoute["routeName"];
-  let emptyStart = '<span class="far fa-star"></span>';
-  routeRating.innerHTML = emptyStart.repeat(5);
+  let emptyStar = '<span class="far fa-star"></span>';
+  let halfStar = '<span class="fas fa-star-half-alt"></span>';
+  let fullStar = '<span class="fas fa-star checked"></span>';
+
+  let ratingCopy = newRoute["rating"];
+  let numberOfFullStars = 0;
+  while (ratingCopy >= 1) {
+    numberOfFullStars += 1;
+    ratingCopy -= 1;
+  }
+  let numberOfHalfStars = 0;
+  if (ratingCopy > 0) {
+    numberOfHalfStars = 1;
+  }
+
+  routeRating.innerHTML =
```

Comment:
If the UI only supports full an half starts, it might be better to have different boundaries for deciding it. At the moment, it will almost always show a half star unless the rating is exactly an integer value. This mean a rating of 1.00001 and 1.9999 will look exactly the same, but 1.9999 and 2.00001 will look very different. Perhaps something like

- if remainder < 0.25, don't add a half star
- if remainder >=0.25 and < 0.75, add a half start
- if remainder >= 0.75 add another full star

Or perhaps 0.2 and 0.8 or something like that.